BLSQ/iaso
# IA-2345: Permissions grouping for IASO admin

Explain what problem this PR is resolving

- Group permissions into modules
- Limit the access of permissions bye modules

Related JIRA tickets : [IA-2345](https://bluesquare.atlassian.net/browse/IA-2345)(Permissions grouping for IASO admin)

## Self proofreading checklist


## Changes

- Added modules and their linked permissions in constants
- Link Account with modules through ArrayField
- Create modules to all existing Accounts
- Add modules JsonField on setup_account
- Edit filter_permissions to take in account modules
- Display on the front end modules(those activated for the account and the other not activated
- Filter on permissions related to current account modules when creating or editing a profile

## How to test
- Go in admin/iaso/account/
- Create an account with some modules
- Create a user with the account
- Check in menu which menus is accessible

## Print screen / video
![Screenshot from 2023-10-17 16-17-06](https://github.com/BLSQ/iaso/assets/19631540/5c8eb189-e386-4a31-892e-93a68ec1e92e)
![Screenshot from 2023-10-17 16-17-44](https://github.com/BLSQ/iaso/assets/19631540/ed8aeb8a-ee18-4d6c-9a8f-b1753d58bb62)
f9d98e3
![Screenshot from 2023-10-17 16-18-27](https://github.com/BLSQ/iaso/assets/19631540/576ba2b0-4dbf-4053-b705-5e81f2ad86b7)
)
![Screenshot from 2023-10-17 16-19-52](https://github.com/BLSQ/iaso/assets/19631540/2d5172cf-5517-4504-ab68-3f826d357f59)

## Notes

Things that the reviewers should know: known bugs that are out of the scope of the PR, other trade-offs that were made.
If the PR depends on a PR in bluesquare-components, or merges into another PR (i.o. main), it should also be mentioned here


[IA-2345]: https://bluesquare.atlassian.net/browse/IA-2345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
